src/routes/(console)/apply-credit/+page.svelte

```diff
@@ -64,136 +55,85 @@
-                    selectedOrg.$id,
-                    billingPlan,
-                    paymentMethodId,
-                    null
-                );
-            }
-            // Existing pro org
-            else {
-                org = selectedOrg;
-            }
+            await sdk.forConsole.billing.addCredit(org.$id, couponData.code);
+            trackEvent(Submit.CreditRedeem);
+        } catch (e) {
+            error = e;
+            $isSubmitting = false;
+            trackError(error, Submit.CreditRedeem);
+        }
 
-            // Add coupon
-            if (couponData?.code) {
-                await sdk.forConsole.billing.addCredit(org.$id, couponData.code);
-            }
+        addNotification({
+            type: error ? 'error' : 'success',
+            message: error ? error.message : 'Credit applied successfully'
+        });
 
-            // Add budget
-            if (billingBudget) {
-                await sdk.forConsole.billing.updateBudget(org.$id, billingBudget, [75]);
-            }
+        if (!error) {
+            await goto(`${base}/organization-${selectedOrgId}`);
+        }
+    }
 
-            // Add collaborators
-            if (collaborators?.length) {
-                collaborators.forEach(async (collaborator) => {
-                    await sdk.forConsole.teams.createMembership(
-                        org.$id,
-                        ['owner'],
-                        collaborator,
-                        undefined,
-                        undefined,
-                        `${$page.url.origin}${base}/invite`
-                    );
-                });
+    async function handleSubmit() {
+        if (couponData) {
+            if (
+                selectedOrg?.billingPlan === BillingPlan.PRO &&
+                campaign.plan !== BillingPlan.SCALE
+            ) {
+                await applyCouponCredit(selectedOrg);
+                return;
             }
 
-            // Add tax ID
-            if (taxId) {
-                await sdk.forConsole.billing.updateTaxId(org.$id, taxId);
-            }
-            trackEvent(Submit.CreditRedeem, {
-                coupon: couponData.code,
-                campaign: couponData?.campaign
-            });
-            await invalidate(Dependencies.ORGANIZATION);
-            await goto(`${base}/organization-${org.$id}`);
-            addNotification({
-                type: 'success',
-                message: 'Credits applied successfully'
-            });
-            await invalidate(Dependencies.ACCOUNT);
-        } catch (e) {
-            trackError(e, Submit.CreditRedeem);
-            addNotification({
-                type: 'error',
-                message: e.message
-            });
-        }
-    }
+            const createOrganization = selectedAction === 'create';
+            const isScalePlanUpgrade =
+                selectedOrg &&
+                (selectedOrg?.billingPlan as Tier) === BillingPlan.PRO &&
```

Comment:
Can we check `campaign?.plan && selectedOrg?.billingPlan !== campaign?.plan` instead.